benchmarks/synth-bm/src/metrics.rs

```diff
@@ -0,0 +1,215 @@
+use std::fmt::Debug;
+use std::str::FromStr;
+use std::time::{Duration, Instant};
+
+use log::info;
+use reqwest::Client;
+use tokio::time;
+use tokio::time::Interval;
+
+/// The response sent by a NEAR node for `/metrics` queries.
+///
+/// Structured text, containing comments prefixed by `#` and lines like
+/// ```
+/// metric_name{optional_specifiers} metric_value
+/// ```
+pub type Report<'a> = &'a str;
+
+/// Defines metrics for which parsing from a `Report` has been implemented.
+#[derive(Copy, Clone, Debug)]
+pub enum MetricName {
+    /// The number of successfull transactions since the node was started.
+    SuccessfulTransactions,
+}
+
+impl MetricName {
+    fn report_name(self) -> &'static str {
+        match self {
+            MetricName::SuccessfulTransactions => "near_transaction_processed_successfully_total",
```

Comment:
One important thing to keep in mind is that this metric is incremented when the transaction successfully converts into a receipt. It does not mean that the receipt is actually getting executed (it might even happen in a different block if the receipt gets delayed.) You will probably want a metric along the lines of `near_action_called_count` (allows determining the rate of actions based on action types even in mixed action workloads, or where receipts have multiple actions!) or receipts (I don't think we have a single metric for all of them together, but shouldn't be too hard to add up the delayed, local, incoming, etc. receipts.)

Reply:
Interesting! We've been using `near_transaction_processed_successfully_total` for that purpose since the ft benchmarking days...

Will look into `near_action_called_count`.

Reply:
AFAIK this is currently the state of the art for determining successful txs in benchmarks. Usage of `synth-bm` is increasing, so I'd like to propose fixing this separately in order to merge this PR soon. Opened #12931.

Reply:
Even if `transaction_processed` doesn't measure what it was expected to, it might still be useful to have support for that metric in the tooling. E.g. to check if all txs could successfully be converted to receipts.

Reply:
I think in your setup you're also getting numbers from `/metrics` and native transfers are a [single action](https://docs.rs/near-primitives/0.28.0/near_primitives/action/struct.TransferAction.html). So you could get the number of successful native transfers from the line in the `/metrics` response that looks like:

```
near_action_called_count{action="Transfer"} 200
```

(The line may be missing when no transfers were processed successfully yet.)